Update Chromedriver to v114

mcmire opened the pull request:
## Explanation

<!--
Thanks for the pull request. Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

* Fixes #12345
* See: #67890
-->

This avoids version mismatch errors when running e2e tests with Chrome locally.

Note that in order to make this change, we have to instruct LavaMoat to ignore two more globals which are used internally by Chromedriver:

- `Proxy`
- `ret_nodes`

The second global definitely seems like something specific to Chromedriver, but the first one — Proxy — is part of EcmaScript. Always ignoring this seems to go against the spirit of LavaMoat. To address this, then, this commit not only upgrades Chromedriver but also adds a guard around these two exceptions — along with other Chromedriver-specific LavaMoat exceptions we've added in the past — so that they only apply for test builds. This ensure that in the production version of MetaMask, access to Proxy is properly sandboxed.

Inspired by https://github.com/MetaMask/metamask-extension/pull/19180.

## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

You should be able to run all of the Chrome e2e tests locally. In addition, CI should pass.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

mcmire committed:
Align with develop

mcmire committed:
Update Chromedriver to v114

This avoids version mismatch errors when running e2e tests with Chrome
locally.

Note that in order to make this change, we have to instruct LavaMoat to
ignore two more globals which are used internally by Chromedriver:

- `Proxy`
- `ret_nodes`

The second global definitely seems like something specific to
Chromedriver, but the first one — Proxy — is part of EcmaScript. Always
ignoring this seems to go against the spirit of LavaMoat. To address
this, then, this commit not only upgrades Chromedriver but also adds a
guard around these two exceptions — along with other
Chromedriver-specific LavaMoat exceptions we've added in the past — so
that they only apply for test builds. This ensure that in the production
version of MetaMask, access to Proxy is properly sandboxed.

Gudahtt commented on development/build/index.js:
```diff
@@ -82,34 +82,38 @@ async function defineAndRunBuildTasks() {
+      'sentry',
+    ];
+
+    if (entryTask === 'test') {
```
There are two entry tasks for test builds: "test" and "testDev".

We also have constants for each of them. I have another PR open now where I've replaced the other entry task references in this file with constants: https://github.com/MetaMask/metamask-extension/pull/19506

mcmire replied:
Ah, okay. I was going off the `shouldScuttle` array above. Your changes make sense. I'll wait until that PR is merged and then rebase this.

Gudahtt commented:
Creating a separate list of scuttling exceptions for test builds is a great idea. However, the current list of supposedly Chromedriver-related scuttling exceptions is a bit suspect. If there was an entry there relied upon in a production build, it wouldn't be caught by CI. As such we should probably do some manual testing on the production build before reducing that list.

mcmire commented:
@Gudahtt Right, fair enough. I can do that.

mcmire committed:
Merge branch 'develop' into upgrade-chromedriver-to-v114

mcmire committed:
Use constants, and add check for testDev

mcmire committed:
Only conditionally add Proxy and ret_nodes; leave the rest for now

mcmire commented:
@Gudahtt Ah sorry I misread your comment. It sounds like reducing the list can be something we do later. For now I've only added the `if` statement around `Proxy` and `ret_nodes`. Hopefully that addresses your concerns.

Gudahtt reviewed (dismissed):
LGTM!

mcmire committed:
Merge branch 'develop' into upgrade-chromedriver-to-v114

mcmire commented on development/build/index.js:
```diff
@@ -89,52 +148,7 @@ async function defineAndRunBuildTasks() {
-        'Promise',
-        'JSON',
-        'Date',
-        'Proxy',
```
It appears that `Proxy` may have slipped into https://github.com/MetaMask/metamask-extension/pull/19293 by accident, so I moved it back out.

mcmire commented:
^ Fixed conflicts and re-pushed.

mcmire committed:
Merge branch 'develop' into upgrade-chromedriver-to-v114

mcmire committed:
Update LavaMoat policies

mcmire commented:
Sorry one more approval if you please ^
